### PR TITLE
remote kill: don't wait for the container to stop

### PR DESCRIPTION
**Pull request by vrothberg**
Invert the branch logic to match the comment.  Docker seems to wait for
the container while Podman does not.

Fixes: #7135
Signed-off-by: Valentin Rothberg <rothberg@redhat.com>

### CODE REVIEW

**Comment by edsantiago** [comment]
Would you mind removing the "FIXME: pending #7135" skip in `test/system/130-kill.bats`? (I know it kills the elegance of a one-character patch, sorry!)

**Comment by vrothberg** [comment]
> Would you mind removing the "FIXME: pending #7135" skip in `test/system/130-kill.bats`? (I know it kills the elegance of a one-character patch, sorry!)

Done!

@jwhonce, the kill-wait code has been added with commit d924494f561b but I found no sign in docker that it would actually wait. Did the behaviour change over time?

**Comment by vrothberg** [comment]
> @jwhonce, the kill-wait code has been added with commit [d924494](https://github.com/containers/podman/commit/d924494f561bb878a2b3a7ce438d87ecb934b5fb) but I found no sign in docker that it would actually wait. Did the behaviour change over time?

Ah, Docker is only waiting on sigkill:
```golang
// ContainerKill sends signal to the container
// If no signal is given (sig 0), then Kill with SIGKILL and wait
// for the container to exit.
// If a signal is given, then just send it to the container and return.
func (daemon *Daemon) ContainerKill(name string, sig uint64) error {
        container, err := daemon.GetContainer(name)
        if err != nil {
                return err
        }

        if sig != 0 && !signal.ValidSignalForPlatform(syscall.Signal(sig)) {
                return fmt.Errorf("The %s daemon does not support signal %d", runtime.GOOS, sig)
        }

        // If no signal is passed, or SIGKILL, perform regular Kill (SIGKILL + wait())
        if sig == 0 || syscall.Signal(sig) == syscall.SIGKILL {
                return daemon.Kill(container)
        }
        return daemon.killWithSignal(container, int(sig))
}
```

**Comment by vrothberg** [comment]
/hold

Let me fix that before

File: pkg/api/handlers/compat/containers.go [hunk 1]
```diff
@@ -169,16 +170,16 @@ func KillContainer(w http.ResponseWriter, r *http.Request) {
...
 	}
 
-	if utils.IsLibpodRequest(r) {
-		// the kill behavior for docker differs from podman in that they appear to wait
-		// for the Container to croak so the exit code is accurate immediately after the
-		// kill is sent.  libpod does not.  but we can add a wait here only for the docker
-		// side of things and mimic that behavior
+	// Docker waits for the container to stop if the signal is 0 or
+	// SIGKILL.
+	if !utils.IsLibpodRequest(r) && (signal == 0 || syscall.Signal(signal) == syscall.SIGKILL) {
```
Comment (rhatdan) [inline pkg/api/handlers/compat/containers.go:182]: Why would signal=0 stop the container?  I though signal==0 means just check if the PID exists?

Comment (vrothberg) [reply pkg/api/handlers/compat/containers.go:182]: `signal == 0` in docker terms means default which in turn is SIGKILL

Comment (edsantiago) [reply pkg/api/handlers/compat/containers.go:182]: That's my understanding too: sig0 is never delivered to the target, it's only used as a check by calling code (for existence, also for EPERM). Good catch.

Comment (rhatdan) [reply pkg/api/handlers/compat/containers.go:182]: Ok.

**Comment by rhatdan** [comment]
LGTM

**Comment by mheon** [comment]
LGTM

**Comment by mheon** [comment]
I'm going to hold/LGTM and hope tests go green.
/hold
/lgtm

**Comment by vrothberg** [comment]
@mheon, can you /lgtm again. I accidentally added another commit on top but Cirrus is smart enough to cache results :)

**Pull request merged**